model/group.cc

```diff
@@ -478,12 +484,12 @@ namespace minsky
     iconWidth=fabs(b.x0-b.x1)/z;
     iconHeight=fabs(b.y0-b.y1)/z;
     // account for margins
-    float l, r;
-    margins(l,r);
+    float l, r, t, bm;
+    margins(l,r,t,bm);
     // rescale contents to fit
     double x0, x1, y0, y1;
     contentBounds(x0,y0,x1,y1);
-    double sx=(fabs(b.x0-b.x1)-z*(l+r))/(x1-x0), sy=fabs(b.y0-b.y1)/(y1-y0);
+    double sx=(fabs(b.x0-b.x1)-z*(l+r))/(x1-x0), sy=(fabs(b.y0-b.y1)-z*(t+bm))/(y1-y0);
```

Comment:
I don't think either the horizontal or vertical computations correctly handles the case where b is smaller than the minimum size z*(l+r) and z*(t+bm) respectively.